### PR TITLE
[NPUW] Align chunked prefill KV cache and inputs to the left

### PR DESCRIPTION
**Pull request by DingZhangIntel**
### Details:
 - Establish the foundational left-aligned storage contract for chunked prefill KV cache, token inputs, position IDs, attention masks, Block-KV copies, and auxiliary per-layer inputs.
 - Preserve existing whole-prefill and generate behavior.
 - This is the first PR in a three-part review series. Merge order: #37882 -> #37883 -> #37884.
 - PR2: [Support left-aligned KV cache in attention consumers](https://github.com/openvinotoolkit/openvino/pull/37883)
 - PR3: [Select final logits for left-aligned chunked prefill](https://github.com/openvinotoolkit/openvino/pull/37884)
 - The follow-up branches are intentionally cumulative. Review only their incremental changes with `git diff KVAlign-foundation..KVAlign-attention` and `git diff KVAlign-attention..KVAlign-logits`.

### Tickets:
 - CVS-190643

### AI Assistance:
 - AI assistance used: yes
 - AI assistance was used for code-path analysis, implementation, and test design. Human validation: the NPU unit-test target was built and 19 foundation tests passed.

### CODE REVIEW

**Comment by DingZhangIntel** [comment]
Addressed both Copilot comments in commit `bddf82e0fe`:

- `copy_per_layer_inputs_chunk_to_left()` now skips the full destination memset when `chunk_tokens == dst_seq_len`; for short chunks it clears only the trailing byte range before copying the left-aligned data.
- `prefill_present()` now validates that the expected `past_key_values` marker exists before calling `replace()`, producing an actionable assertion for malformed test KV names.

The same fix was propagated to the dependent stacked branches `KVAlign-attention` and `KVAlign-logits`. Focused foundation validation passed: 18/18 tests.

**Comment by DingZhangIntel** [comment]
Addressed the latest blocker in commit `a204a9777c`.

During continued prefill, the caller-provided `token_type_ids` tensor contains only the delta tokens, while `num_stored_tokens` includes the retained prefix. The staging path now uses the delta-relative offset:

```cpp
num_stored_tokens - m_continued_prefill_base
```

This matches the existing input_ids, position_ids, and per_layer_inputs handling. The same commit was propagated to #37883 and #37884. Focused continued-prefill, Block-KV, and per-layer-input tests pass: 18/18.

The generate-only Conv mask rewrite remains in PR2 (`e3c0994959`); PR1 does not duplicate the LFM2-specific construct-level fixture because the synthetic unit model cannot faithfully reproduce the exported Conv IR.

**Comment by DingZhangIntel** [comment]
Hi @AlexanderKalistratov Comments from copilot have been addressed, could you help take a look at PR?

**Comment by DingZhangIntel** [comment]
Moved to the fork integration flow: https://github.com/DingZhangIntel/openvino/pull/1`n`nThe replacement PR targets kv_cache_integration_branch for combined review and full validation before the final upstream PR to master.
